docs/Upgrade.md

```diff
@@ -759,6 +759,60 @@ Besides, the buttons passed as `bulkActionButtons` no longer receive any prop. I
 
 The deprecated `<PaginationLimit>` component was removed.
 
+### `<DatagridBody>` No Longer Provides `record` Prop To `<DatagridRow>`
+
+The `<DatagridBody>` component no longer provides a `record` prop to its `<DatagridRow>` children. Instead, it provides a `recordContext` for each row: 
+
+```diff
+const DatagridBody: FC<DatagridBodyProps> = React.forwardRef(({/*...*/}, ref) 
+    => (
+        <TableBody>
+-            {data.map((record, rowIndex) =>
+-                cloneElement(
++            {data.map((record, rowIndex) => (
++               <RecordContextProvider
++                   value={record}
++                   key={record.id ?? `row${rowIndex}`}
++               >
++                   {cloneElement(
+                        row,
+                        {
+                            //...
+                            id: record.id ?? `row${rowIndex}`,
+-                           key: record.id ?? `row${rowIndex}`,
+                            onToggleItem,
+-                           record,
+                            resource,
+                            //...
+                        },
+                        children
+-                   )
+                    )}
++               </RecordContextProvider>
```

Comment:
wrong diff: you should show how a custom DatagridRow no longer reads the record from its props, but from the Recordcontext.